lib/timescaledb/rails/models/application_record.rb

```diff
@@ -0,0 +1,24 @@
+# frozen_string_literal: true
+
+module Timescaledb
+  module Rails
+    # :nodoc:
+    class ApplicationRecord < ::ActiveRecord::Base
+      self.abstract_class = true
+
+      def self.timescale_connection?(connection)
+        pool_name = lambda do |pool|
+          if pool.respond_to?(:db_config)
+            pool.db_config.name
+          elsif pool.respond_to?(:spec)
+            pool.spec.name
+          else
+            raise "Don't know how to get pool name from #{pool.inspect}"
+          end
+        end
+
+        pool_name[connection.pool] == pool_name[self.connection.pool] && Hypertable.table_exists?
```

Comment:
This looks much better now, thank you 💯 . I would remove the `Hypertable.table_exists?` check here, I'd only check the connection here.

Reply:
So I think the issue with that is if someone has the gem installed and has not configured a custom connection for it -- we will assume that the primary conenction is for timescale, even if it is e.g. postgres

Reply:
Reading my comment again, I was not clear enough, sorry about that. My suggestion was to move it back to where it was before because it seems weird to have a reference of an ApplicationRecord child here in the parent, what do you think? We would only check the connection here (which based on the name makes sense) and then  in the timescale_enabled? helper method, we would check the connection + if the hypertable table exists, sounds ok to you?
```ruby
def timescale_enabled?
  ApplicationRecord.timescale_connection?(connection) && Timescaledb::Rails::Hypertable.table_exists?
end
```

Reply:
Ah yup that makes sense. Can make that change tonight.

Reply:
I'll try to publish a new version as soon as we get this merged 💪

Reply:
@segiddins Let me know if there is anything I can help you with, I'd love to merge this and publish a new version 💯